### ISSUE TITLE
Style/ConcatArrayLiterals autocorrection isn't working

### ISSUE THREAD

**Issue by lucascaton** [rubocop/rubocop#11325]
## Expected behavior

Docs says this cop supports autocorrection:

> <img width="380" alt="image" src="https://user-images.githubusercontent.com/86755/209411583-15735eb4-0373-4005-9570-1d1d8b495b5e.png">

## Actual behavior

Autocorrection is not working (not even with `-A` flag):

```bash
$ rubocop -A

# (...)

Offenses:

file.rb:33:17: C: Style/ConcatArrayLiterals: Use push with elements as arguments without array brackets instead of concat(%i[subject preview_text audience body sender_id]).
          attrs.concat(%i[subject body])
                ^^^^^^^^^^^^^^^^^^^^^^^^

162 files inspected, 1 offense detected
```

## Steps to reproduce the problem

Here's the tested Ruby code:

```ruby
attrs.concat(%i[subject body])
```

## RuboCop version

```
$ rubocop -V
1.41.1 (using Parser 3.1.3.0, rubocop-ast 1.24.0, running on ruby 3.1.3) [arm64-darwin22]
  - rubocop-performance 1.15.1
  - rubocop-rails 2.17.3
  - rubocop-rspec 2.16.0
```
